shared_model/bindings/model_transaction_builder_embind.cpp

```diff
@@ -0,0 +1,143 @@
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+
+#include "model_proto.hpp"
+#include "model_transaction_builder.hpp"
+
+using namespace emscripten;
+using namespace shared_model;
+using namespace shared_model::bindings;
+using namespace shared_model::proto;
+
+EMSCRIPTEN_BINDINGS(model_transaction_builder)
+{
+  class_<Transaction>("Transaction")
+  .function("creatorAccountId", &Transaction::creatorAccountId)
+  .function("quorum", &Transaction::quorum)
+  .function("blob", &Transaction::blob)
+  .function("payload", &Transaction::payload)
+  .function("addSignature", &Transaction::addSignature)
+  .function("createdTime", &Transaction::getCreatedTime)
+  ;
+
+  typedef UnsignedWrapper<Transaction> UnsignedWrapper;  
+  class_<UnsignedWrapper>("UnsignedTx")
+  .function("signAndAddSignature", &UnsignedWrapper::signAndAddSignature)
+  .function("finish", &UnsignedWrapper::finish)
+  .function("hash", &UnsignedWrapper::hash)
+  ;
+
+  typedef ModelProto<UnsignedWrapper> ModelProto;
+  class_<ModelProto>("ModelProtoTransaction")
+  .constructor<UnsignedWrapper&>()
+  .function("signAndAddSignature", &ModelProto::signAndAddSignature)
+  .function("finish", &ModelProto::finish)
+  ;
+
+  using Role = interface::permissions::Role;
+  enum_<Role>("Role")
+  .value("kAppendRole", Role::kAppendRole)
```

Comment:
Can initialization and type adding be splitted? e.g:
```cpp
auto e = enum_<Role>("Role");
e.value(..., ...);
```

If so, than it's possible to use smth like:
```cpp
auto e = enum_<Role>("Role");
for (auto i = 0; i < Role::COUNT; ++i) {
  auto p = static_cast<Role>(i);
  e = e.value(toString(p), p);
}
```